### ISSUE TITLE
Question: why yet another montool?

### ISSUE THREAD

**Issue by faxm0dem** [sebt3/watched.agent#1]
Hi,

What is the main motivation of your project? There are so many tools out there, and does the world need yet another? Maybe yes! But then again I don't see the main differences of your project as compared to others.

Cheers

**Comment by sebt3** [comment]
Thanks for your interest :)
There's indeed tons of solution for monitoring. Zabbix, Nagios and so many more. But most these solutions need lot's of configuration to get going. This agent detect itself what need to be watched and start collecting metrics once started without any initial configuration. (A configuration file is still there to allow you to disable some monitoring/tune configuration).
Beside, most solution are script based and have a large overhead per metric collection. This agent is trying to keep the running cost low to allow more frequent metric collection (having most metric collected on a 5s interval, while maintaining a performance cost bellow what a standard Nagios configuration can get's you). That, later, will allow a more in depth performance analytics.
Then I kind of dislike RRDtools. It's a good storage system for performance metrics, yet it's hard to use it to, say, compare 2 services performances. Custom reports are a pain to say at least. Having a general purpose graphing tool and a mysql backend make custom reports a breath.
Finally having a standard architecture (PHP/mysql front-end) and a scalable backend will allow users of the solution go rather crazy on the kind of deployment scenarios they want to have. Something like : https://sebt3.github.io/watched/screens/watched_architecture.png is already completly possible.

**Comment by faxm0dem** [comment]
How would you compare it to collectd or netdata?

**Comment by sebt3** [comment]
Well both are production ready which is not the case of my project (just yet).
Collectd send the collected data to RRD files. and dont allow for much data analyzis (beside what RRD provide, which, to me, is unsatifactory)
I discovered netdata after starting my project. Interesting. But it's more for real time stuff than anything else. So once again it's failing on the long term analysis I want to have

**Comment by faxm0dem** [comment]
You're wrong about collectd: RRD is only *one* of *many* possible write plugins (backends).
Most collectd users today store their metrics in influxdb, elasticsearch, opentsdb, graphite, etc.
